### PR TITLE
Creating contract-tests for basic SELECT query with MySQL

### PR DESCRIPTION
**Pull request by ektabj**
*Issue #, if available:*
N/A

*Description of changes:*
Creating contract-tests for basic SELECT query with MySQL.

Testing:

```
<whatever>: pytest contract-tests/tests

================================================================ test session starts =================================================================
platform linux -- Python 3.12.3, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/ektabj/aws-otel-python-instrumentation/contract-tests/tests
configfile: pyproject.toml
collected 2 items                                                                                                                                    

../../../../home/ektabj/aws-otel-python-instrumentation/contract-tests/tests/test/amazon/pymysql/pymysql_test.py::PyMysqlTest::test_fault 
------------------------------------------------------------------- live log setup -------------------------------------------------------------------
2024-06-12 19:13:37 [    INFO] Pulling image aws-application-signals-mock-collector-python (container.py:53)
2024-06-12 19:13:37 [    INFO] Container started: d7238d399968 (container.py:64)
2024-06-12 19:13:38 [    INFO] Pulling image mysql:latest (container.py:53)
2024-06-12 19:13:39 [    INFO] Container started: 18c0eaa21b11 (container.py:64)
2024-06-12 19:13:39 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:39 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:40 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:41 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:42 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:43 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:44 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:45 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:46 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:47 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:48 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:49 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:50 [    INFO] Waiting to be ready... (waiting_utils.py:46)
------------------------------------------------------------------- live log call --------------------------------------------------------------------
2024-06-12 19:13:50 [    INFO] Pulling image aws-application-signals-tests-pymysql-app (container.py:53)
2024-06-12 19:13:50 [    INFO] Container started: 253735cdb4d2 (container.py:64)
2024-06-12 19:13:56 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:57 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:13:57 [    INFO] Application stdout (contract_test_base.py:118)
2024-06-12 19:13:57 [    INFO] Ready
Expected Exception with Invalid SQL occurred: (1146, "Table 'testdb.invalid_table' doesn't exist")
 (contract_test_base.py:119)
2024-06-12 19:13:57 [    INFO] Application stderr (contract_test_base.py:120)
2024-06-12 19:13:57 [    INFO] Failed to get k8s token: [Errno 2] No such file or directory: '/var/run/secrets/kubernetes.io/serviceaccount/token'
AwsEcsResourceDetector failed: Missing ECS_CONTAINER_METADATA_URI therefore process is not on ECS.
AwsEksResourceDetector failed: [Errno 2] No such file or directory: '/var/run/secrets/kubernetes.io/serviceaccount/token'
Exception  in detector <opentelemetry.sdk.extension.aws.resource.ec2.AwsEc2ResourceDetector object at 0x7fc36a868df0>, ignoring
AwsEc2ResourceDetector failed: timed out
Configuration of configurator not loaded, aws_configurator already loaded
 (contract_test_base.py:121)
PASSED                                                                                                                                         [ 50%]
../../../../home/ektabj/aws-otel-python-instrumentation/contract-tests/tests/test/amazon/pymysql/pymysql_test.py::PyMysqlTest::test_success 
------------------------------------------------------------------- live log call --------------------------------------------------------------------
2024-06-12 19:13:57 [    INFO] Pulling image aws-application-signals-tests-pymysql-app (container.py:53)
2024-06-12 19:13:58 [    INFO] Container started: 57ab71887ac6 (container.py:64)
2024-06-12 19:14:04 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:14:04 [    INFO] Waiting to be ready... (waiting_utils.py:46)
2024-06-12 19:14:04 [    INFO] Application stdout (contract_test_base.py:118)
2024-06-12 19:14:04 [    INFO] Ready
 (contract_test_base.py:119)
2024-06-12 19:14:04 [    INFO] Application stderr (contract_test_base.py:120)
2024-06-12 19:14:04 [    INFO] AwsEcsResourceDetector failed: Missing ECS_CONTAINER_METADATA_URI therefore process is not on ECS.
Failed to get k8s token: [Errno 2] No such file or directory: '/var/run/secrets/kubernetes.io/serviceaccount/token'
AwsEksResourceDetector failed: [Errno 2] No such file or directory: '/var/run/secrets/kubernetes.io/serviceaccount/token'
Exception  in detector <opentelemetry.sdk.extension.aws.resource.ec2.AwsEc2ResourceDetector object at 0x7f21390ecd00>, ignoring
AwsEc2ResourceDetector failed: timed out
Configuration of configurator not loaded, aws_configurator already loaded
 (contract_test_base.py:121)
PASSED                                                                                                                                         [100%]
----------------------------------------------------------------- live log teardown ------------------------------------------------------------------
2024-06-12 19:14:05 [    INFO] MockCollector stdout (contract_test_base.py:71)
2024-06-12 19:14:05 [    INFO] Ready
 (contract_test_base.py:72)
2024-06-12 19:14:05 [    INFO] MockCollector stderr (contract_test_base.py:73)
2024-06-12 19:14:05 [    INFO]  (contract_test_base.py:74)


================================================================== warnings summary ==================================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================================== 2 passed, 2 warnings in 30.13s ===========================================================

```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/set-up-contract-tests.sh [hunk 1]
```diff
@@ -18,6 +18,9 @@ rm -rf dist/mock_collector*
 rm -rf dist/contract_tests*
 
 # Install python dependency for contract-test
+pip install pymysql
```
[the remark this answers is not in the document]
Comment (ektabj) [reply scripts/set-up-contract-tests.sh:21]: Yes, `pymysql` is required by `MySqlContainer` when working with Testcontainers in Python.

File: contract-tests/tests/test/amazon/pymysql/pymysql_test.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    AWS_REMOTE_SERVICE,
+    AWS_SPAN_KIND,
+)
+from opentelemetry.proto.common.v1.common_pb2 import AnyValue, KeyValue
+from opentelemetry.proto.metrics.v1.metrics_pb2 import ExponentialHistogramDataPoint, Metric
+from opentelemetry.proto.trace.v1.trace_pb2 import Span
+from opentelemetry.trace import StatusCode
+
+
+class PyMysqlTest(ContractTestBase):
```
[the remark this answers is not in the document]
Comment (ektabj) [reply contract-tests/tests/test/amazon/pymysql/pymysql_test.py:25]: Yes, but we will do it as part of a different task.

File: scripts/set-up-contract-tests.sh [hunk 3]
```diff
@@ -18,6 +18,9 @@ rm -rf dist/mock_collector*
...
+pip install cryptography
```
[the remark this answers is not in the document]
Comment (ektabj) [reply scripts/set-up-contract-tests.sh:22]: `cryptography` is used to ensure security and data integrity. 
Without `cryptography`, build will fail with error: `RuntimeError: 'cryptography' package is required for sha256_password or caching_sha2_password auth methods`.

File: contract-tests/tests/test/amazon/pymysql/pymysql_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thpierce) [reply contract-tests/tests/test/amazon/pymysql/pymysql_test.py:25]: Alright.

File: scripts/set-up-contract-tests.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (thpierce) [reply scripts/set-up-contract-tests.sh:21]: Makes sense. I think Psycopg has the same deal.

File: scripts/set-up-contract-tests.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (thpierce) [reply scripts/set-up-contract-tests.sh:22]: Weird, we don't call those methods in your code. Seems like an optional/hidden dependency of `pymysql`: https://stackoverflow.com/questions/54477829/cryptography-is-required-for-sha256-password-or-caching-sha2-password

Anyways, seems fine.

**Pull request merged**